#VRML_SIM R2025a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/robots/gctronic/e-puck/protos/E-puck.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2025a/projects/appearances/protos/ThreadMetalPlate.proto"

WorldInfo {
}
Viewpoint {
  orientation 0.34118161187859436 -0.1964806781392109 -0.919233621466208 4.115319120976626
  position 2.1271337818277396 -2.7924509531183563 2.9968976585623808
}
TexturedBackground {
  skybox FALSE
  skyColor [
    0.304494 0.304494 0.304494
  ]
}
TexturedBackgroundLight {
  luminosity 0
}
DEF ARENA RectangleArena {
  name "arena"
  floorSize 1 1.5
  floorAppearance ThreadMetalPlate {
    type "square"
  }
}
DEF ROBOT1 E-puck {
  hidden position_0_0 59.77997009138721
  hidden position_0_1 49.21589133809031
  hidden rotation_1 0 -1 0 3.0518829804086516
  hidden rotation_2 0 -1 0 1.0495911193463832
  translation 0.401214 -0.601868 -8.831e-05
  rotation 0.0006503828169380304 -0.0003740828947076249 0.9999997185320502 3.1366
  name "robot1"
  controller "main_controller"
  supervisor TRUE
  groundSensorsSlot [
    DEF GROUND_LIGHT_ROBOT1 PointLight {
      attenuation 0 0 12
      color 0.0985428 0.931792 0.216831
      location -0.01 0 0
    }
  ]
}
DEF ROBOT2 E-puck {
  hidden position_0_0 56.97475706423958
  hidden position_0_1 44.334629920381296
  hidden rotation_1 0 -0.9999999999999999 0 5.857096007556281
  hidden rotation_2 0 -1 0 5.9308525370553955
  translation 0.393193 -0.412757 -5.09988e-05
  rotation -0.0009721877309806135 0.0012681796490750703 0.9999987232848816 -3.113635307179586
  name "robot2"
  controller "main_controller"
  controllerArgs [
    ""
  ]
  customData "[0.21757110373718413, -0.040868443024398765, -0.347865103489809, -0.21587933009098895, -0.3319798072598702, -0.16637086861246944, 0.29476899141522384, 0.2913741026652337, 0.20739118101042903, 0.17819137012496933, 0.05431553651331259, -0.1771127711795776, -0.25986037766206627, 0.05913979940606584, -0.058937116753271714, 0.1867470897784537, -0.19285699353333197, -0.1013679580536467, -0.16687946885921723, 0.11478503826343109, -0.2684242035392152, -0.21154140717957381, -0.026819979884992184, -0.10345356034040007, -0.27647766393771106, -0.18039283460765454, 0.20681246083869226, -0.1539840606938902, -0.25356236235217866, 0.23336092535311484, 0.4103006528813425, 0.11481508539562524, -0.33219943846060973, -0.1012781814707186, 0.5829236091432642, -0.2644858163555985, -0.0457180925286515, -0.07089413663227703, -0.023374408894119673, -0.08217452744388881, 0.017228851411175097, -0.007654731456230386, 0.17531181251184916, 0.019297467553386988, 0.1643143538585993, 0.7020026078622412, 0.24912625757244283, 0.07707496078773085, -0.39426067065175957, 0.19043949705404684, -0.1355761551813992, -0.19973525766317257, -0.12730961533449314, -0.1633881608343079, -0.09024079509871555, -0.10918164920853096, 0.4856399905880792, 0.1528900496825959, -0.2841709666951051, -0.07855313587468266, -0.4535700810455331, -0.25666607003293396, -0.05343428673783518, -0.07671429388089379, -0.06396560134367969, -0.0799332261952625, 0.18182447031179044, 0.028590990722116155, 0.018090852911818497, -0.03361725908506127, 0.08582297578240497, -0.08012371022815087, 0.2513531042327177, 0.019880576568443303]"
  supervisor TRUE
  groundSensorsSlot [
    DEF GROUND_LIGHT_ROBOT2 PointLight {
      attenuation 0 0 12
      color 0.0985 0.932 0.217
      location -0.01 0 0
    }
  ]
}
DEF AREA Solid {
  translation -0.296063 0.543857 0.01
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0.155306 0.236423 1
      }
      geometry Plane {
        size 0.4 0.4
      }
    }
  ]
}
DEF OBJECT Solid {
  translation 0.149677 -0.496533 0.07
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0.278813 0.154742
      }
      geometry Box {
        size 0.2 0.3 0.15
      }
    }
  ]
  name "solid(2)"
  boundingObject Box {
    size 0.2 0.3 0.15
  }
  physics Physics {
    mass 0.2
  }
}
